configure.ac

```diff
@@ -1007,39 +917,25 @@ SQUID_DEFINE_BOOL(ICAP_CLIENT,${enable_icap_client:=yes}, [Enable ICAP client fe
 AS_IF(test "x$enable_icap_client" != "xno", squid_opt_use_adaptation="yes")
 AM_CONDITIONAL(ENABLE_ICAP_CLIENT, test "x$enable_icap_client" != "xno")
 
-squid_opt_use_ecap=1
-AC_MSG_CHECKING(whether to support eCAP)
 AC_ARG_ENABLE(ecap,
-  AS_HELP_STRING([--enable-ecap],[support loadable content adaptation modules]), [
-    case "${enableval}" in
-       yes|no) squid_opt_use_ecap=$enableval ;;
-       *) AC_MSG_ERROR(bad value ${enableval} for --enable-ecap) ;;
-    esac
-AC_MSG_RESULT([$squid_opt_use_ecap, explicitly])
-    ], [
-        squid_opt_use_ecap=no;
-        AC_MSG_RESULT([$squid_opt_use_ecap, implicitly])
-    ]
-)
-
+  AS_HELP_STRING([--enable-ecap],[support loadable content adaptation modules]),[
+  SQUID_YESNO([$enableval],[bad value $enableval for --enable-ecap])
+])
 dnl Perform configuration consistency checks for eCAP
-if test "x$squid_opt_use_ecap" != "xno";
```

Comment:
The official AC_ARG_ENABLE( ) code leaves squid_opt_use_ecap set to either "yes" or "no". The implicit/default outcome is "no".

AC_ARG_ENABLE() code in the PR does not have an implicit handling clause, which leaves  enable_ecap in one of the _three_ states. That, in turn, breaks the logic behind `... != "xno"` comparisons below, leading to failed default [builds](https://semaphoreci.com/squid/squid/branches/pull-request-937/builds/8). AFAICT, after PR changes, all the `enable_ecap != xX` comparisons should be written in a direct manner (e.g., `enable_ecap = xY`) to work correctly when enable_ecap has three possible states/values.

The PR no longer logs helpful "implicitly"/"explicitly" notes when checking whether to enable eCAP support. I do not insist on their preservation.